base/reflection.jl

```diff
@@ -74,6 +74,13 @@ end
 fieldnames(t::DataType) = Symbol[fieldname(t, n) for n in 1:nfields(t)]
 fieldnames{T<:Tuple}(t::Type{T}) = Int[n for n in 1:nfields(t)]
 
+"""
+    datatype_module(dt) -> Module
+
+Determine the module containing the definition of a `DataType`.
+"""
+datatype_module(dt) = typeof(dt).name.module
```

Comment:
My interpretation of the docstring is that `dt` is the type in question and not an instance of it, e.g. `datatype_module(Range)` should yield `Base`. But then the `typeof` is too much. (`typeof(Range).name.module` would be `DataType.name.module`, namely `Core`).